client/state/data-layer/wpcom/user-settings/index.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { isEmpty, keys, merge, noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { decodeEntities } from 'lib/formatting';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { updateUserSettings, clearUnsavedUserSettings } from 'state/user-settings/actions';
+import {
+	USER_SETTINGS_REQUEST,
+	USER_SETTINGS_SAVE,
+} from 'state/action-types';
+
+/*
+ * Decodes entities in those specific user settings properties
+ * that the REST API returns already HTML-encoded
+ */
+function fromApi( apiResponse ) {
+	const decodedValues = {
+		display_name: apiResponse.display_name && decodeEntities( apiResponse.display_name ),
+		description: apiResponse.description && decodeEntities( apiResponse.description ),
+		user_URL: apiResponse.user_URL && decodeEntities( apiResponse.user_URL )
+	};
+
+	return merge( {}, apiResponse, decodedValues );
+}
+
+export const requestUserSettings = ( { dispatch }, action, next ) => {
+	dispatch( http( {
+		apiVersion: '1.1',
+		method: 'GET',
+		path: '/me/settings',
+	}, action ) );
+
+	return next( action );
+};
+
+export const receiveUserSettings = ( { dispatch }, action, next, data ) => {
+	dispatch( updateUserSettings( fromApi( data ) ) );
+};
+
+export const receiveRequestError = noop;
+
+/*
+ * Post settings to WordPress.com API at /me/settings endpoint
+ */
+export function saveUserSettings( { dispatch, getState }, action, next ) {
+	const { settingsOverride } = action;
+	const settings = settingsOverride || getState().userSettings.unsavedSettings;
+
+	if ( ! isEmpty( settings ) ) {
+		dispatch( http( {
+			apiVersion: '1.1',
+			method: 'POST',
+			path: '/me/settings',
+			body: settings,
+		}, action ) );
+	}
+
+	return next( action );
+}
+
+export const receiveSavedUserSettings = ( { dispatch }, { settingsOverride }, next, data ) => {
```

Comment:
there's no reason to block on the name here, but still it's obvious from the code that it's actually receiving the response but that doesn't really clue us in to its purpose. the reason I suggested a name change is so that the name can convey the intention of this function the same way we want to indicate the _why_ in a comment more than the _what_, since the _what_ is often obvious but the _why_ less so